File: Introduccion_a_POO/Ejercicio7/Mascota.java
package PRACTICA1AUXILIATURA.Introduccion_a_POO.Ejercicio7;

public class Mascota {
    private String nombre;
    private String tipo;
    private int energia;

    public Mascota(String nombre, String tipo, int energia){
        this.nombre = nombre;
        this.tipo = tipo;
        this.energia = energia;
    }
    public void alimentar(){
        if (energia >= 81){
            System.out.println("Si alimenta ahora a su mascota, sobrepasara el limite de energia, por favor juegue con ella primero.");
        } else{     
            energia = energia + 20;
            System.out.println(nombre + " ha sido alimentado, su energia es: " + energia);
        }
    }
    public void jugar(){
        if (energia >= 15){
            energia = energia - 15;
            System.out.println(nombre + " ha jugado, su energia es: " + energia);
        } else{
            System.out.println(nombre + " no tiene suficiente energia para jugar, su energia es de: " + energia + " por favor aliméntelo primero.");
        }
    }
    public String toString(){
        return "Nombre: " + nombre + ", Tipo: " + tipo + ", Energia: " + energia;
    }

    public static void main(String[] args){
        Mascota m1 = new Mascota("Firulais","Perro",50);
        Mascota m2 = new Mascota("Pelusa","Gato",30);
        System.out.println("Al inicio: " + m1.toString());
        m1.jugar();
        m1.alimentar();
        m1.alimentar();
        m1.jugar();
        System.out.println("Al final: " + m1.toString());
        System.out.println("-------------------");
        System.out.println("Al inicio: " + m2.toString());
        m2.jugar();
        m2.jugar();
        m2.jugar();
        m2.alimentar();
        m2.jugar();
        System.out.println("Al final: " + m2.toString());
    }
}

File: Introduccion_a_POO/Ejercicio7/Mascota.py
class Mascota:
    def __init__(self, nombre, tipo, energia):
        self.nombre = nombre
        self.tipo = tipo
        self.energia = energia

    def alimentar(self):
        if self.energia >= 81:
            print("Si alimenta ahora a su mascota, sobrepasara el limite de energia, por favor juegue con ella primero.")
        else:
            self.energia += 20
            print(f"{self.nombre} ha sido alimentado, su energia es: {self.energia}")

    def jugar(self):
        if self.energia >= 15:
            self.energia -= 15
            print(f"{self.nombre} ha jugado, su energia es: {self.energia}")
        else:
            print(f"{self.nombre} no tiene suficiente energia para jugar, su energia es de: {self.energia} por favor aliméntelo primero.")

    def __str__(self):
        return f"Nombre: {self.nombre}, Tipo: {self.tipo}, Energia: {self.energia}"


# Main
if __name__ == "__main__":
    m1 = Mascota("Firulais", "Perro", 50)
    m2 = Mascota("Pelusa", "Gato", 30)

    print("Al inicio: " + str(m1))
    m1.jugar()
    m1.alimentar()
    m1.alimentar()
    m1.jugar()
    print("Al final: " + str(m1))
    print("-------------------")
    print("Al inicio: " + str(m2))
    m2.jugar()
    m2.jugar()
    m2.jugar()
    m2.alimentar()
    m2.jugar()
    print("Al final: " + str(m2))
    print("-------------------")